spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/data/vectorized/ColumnarBatchReader.java

```diff
@@ -245,5 +269,20 @@ void applyEqDelete(ColumnarBatch columnarBatch) {
 
       columnarBatch.setNumRows(currentRowId);
     }
+
+    ColumnarBatch removeExtraColumns(
+        ColumnVector[] arrowColumnVectors, ColumnarBatch columnarBatch) {
+      int numOfExtraColumns = numOfExtraColumns(deletes);
+      if (numOfExtraColumns > 0) {
+        int newLength = arrowColumnVectors.length - numOfExtraColumns;
+        // In DeleteFilter.fileProjection, the columns for missingIds (the columns required
+        // for equality delete or ROW_POSITION) are appended to the end of the expectedSchema.
+        // Therefore, these extra columns can be removed from the end of arrowColumnVectors.
+        ColumnVector[] newColumns = Arrays.copyOf(arrowColumnVectors, newLength);
+        return new ColumnarBatch(newColumns, columnarBatch.numRows());
+      } else {
+        return columnarBatch;
+      }
+    }
```

Comment:
How about refactor it like this,  so that method `numOfExtraColumns` is not needed? Can we also move all related comments to this method's Java doc?
```
      int expectedColumnSize = deletes.expectedSchema().columns().size();
      if (arrowColumnVectors.length > expectedColumnSize) {
        ColumnVector[] newColumns = Arrays.copyOf(arrowColumnVectors, expectedColumnSize);
        return new ColumnarBatch(newColumns, columnarBatch.numRows());
      } else {
        return columnarBatch;
      }

```

Reply:
Good suggestion! Changed.